tests/integration/commands/deploy/deploy.test.js

```diff
@@ -27,9 +27,10 @@ const validateContent = async ({ content, path, siteUrl, t }) => {
     }
     t.expect(body).toEqual(content)
   } catch (error) {
-    const {
-      response: { statusMessage },
-    } = error
+    let statusMessage = 'Unable to retrieve status message'
+    if (typeof error.response !== 'undefined') {
+      ;({ statusMessage } = error.response)
+    }
```

Comment:
I've been getting failures here to extract `statusMessage` because `error.response` was undefined. After this change now I'm seeing:

https://github.com/netlify/cli/actions/runs/7958720805/job/21724155516#step:11:1309
```
 FAIL  tests/integration/commands/deploy/deploy.test.js > commands/deploy > should deploy Edge Functions when directory exists
Error: Failed getting content: 504 - Unable to retrieve status message - 
 ❯ validateContent tests/integration/commands/deploy/deploy.test.js:35:11
```

https://github.com/netlify/cli/actions/runs/7958720805/job/21724155516#step:11:1320
```
 FAIL  tests/integration/commands/deploy/deploy.test.js > commands/deploy > should deploy Edge Functions with custom cwd when directory exists
Error: Failed getting content: 502 - Unable to retrieve status message - The subhoster origin returned a failure status for the boot RPC call. Status: 502 Bad Gateway
 ❯ validateContent tests/integration/commands/deploy/deploy.test.js:35:11
```

Is this at all related to changes I'm attepting in this PR? It wouldn't seem so to me because those tests don't seem related to images and I don't think they even use code paths I'm changing, but I'm not 100% sure here

Reply:
Those failures do seem like "flakes" - but at least this change exposes underlying problems instead of "cannot read statusMessage of undefined" (to make it easier for future)

Reply:
This was changed a bit in https://github.com/netlify/cli/pull/6385 so I just used that when resolving conflicts

Reply:
Yeah it seems fine now :)